odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,452 @@
+
+    def __init__(self, space, constant):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `LinearSpace`
+            Domain of the functional.
+        constant : element in `domain.field`
+            The constant value of the functional
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=0)
+
+        if constant not in self.range:
+            raise TypeError('constant {} not in the range {}.'
+                            ''.format(constant, self.range))
+
+        self.__constant = constant
+
+    @property
+    def constant(self):
+        return self.__constant
+
+    def _call(self, x):
+        """Return a constant value."""
+        return self.constant
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional."""
+        return ZeroOperator(self.domain)
+
+    @property
+    def proximal(self):
+        """Return the proximal factory of the functional."""
+        return proximal_zero(self.domain)
+
+    # TODO: Update name for proximal_zero. It is the proximal for any constant
+    # functional
+
+    @property
+    def convex_conj(self):
+        """Convex conjugate functional of the constant functional.
+
+        Notes
+        -----
+        This functional is defined as
+
+         .. math::
+
+            f^*(x) = \\left\{ \\begin{array}{ll}
+            -constant & \\text{if } x = 0, \\\\
+            \\infty & \\text{else}
+            \\end{array} \\right.
+        """
+        functional = self
+
+        class ConstantFunctionalConvexConj(Functional):
+
+            """The convex conjugate functional of the constant functional.
+
+            It does not implement `gradient` since it is not differentiable
+            anywhere.
+            """
+
+            def __init__(self):
+                """Initialize a new instance."""
+                super().__init__(functional.domain, linear=False)
+
+            def _call(self, x):
+                """Apply the functional to the given point."""
+
+                if x.norm() == 0:
+                    # In this case x is the zero-element.
+                    return -functional.constant
+                else:
+                    return np.inf
+
+            @property
+            def convex_conj(self):
+                """The convex conjugate functional.
+
+                Notes
+                -----
+                By the Fenchel-Moreau theorem the convex conjugate functional
+                is the constant functional [BC2011]_.
+                """
+                return functional
+
+            @property
+            def proximal(self):
+                """Return the proximal factory of the functional.
+
+                This is the zero operator.
+                """
+                return lambda sigma: ZeroOperator(self.domain)
+
+        return ConstantFunctionalConvexConj()
+
+
+class ZeroFunctional(ConstantFunctional):
+
+    """The zero-functional.
```

Comment:
Could make this a one liner:

"""Functional that maps all elements in the domain to zero."""